packages/babel-plugin-relay/printGraphQL.js

```diff
@@ -0,0 +1,184 @@
+/**
+ * Inlined GraphQL printer,
+ * forked from https://github.com/graphql/graphql-js/blob/v15.3.0/src/language/printer.js
+ *
+ * This ensure compatibility with document hash generated by the Relay compiler.
+ * graphql-js' printer is incompatible with Relay's one since v15.4
+ *
+ * TODO: Canonicalize printer spec for Relay projects
+ *
+ * @see https://github.com/facebook/relay/pull/3628
+ * @see https://github.com/facebook/relay/issues/4226
+ */
+
+const { visit } = require('graphql');
+
+/**
+ * Converts an AST into a string, using one set of reasonable
+ * formatting rules.
+ */
+function print(ast: any): string {
+  return visit(ast, { leave: printDocASTReducer });
```

Comment:
Something weird here. By version 16.8.1 which we are supposed to be using, this had changed and there wasn't a top-level `leave` here anymore, each child visitor had a `leave` key.

I think there's some weird Yarn issues, becuase if I run `yarn` in this directory, I get a local `node_modules` directory with `graphql@16.8.1` which causes this to fail with many failures that look like this:

```
  ● printGraphQL › tests printer idempotence: conditions.expected

    TypeError: visitFn.call is not a function

      19 |  */
      20 | function print(ast: any): string {
    > 21 |   return visit(ast, { leave: printDocASTReducer });
         |                             ^
      22 | }
      23 |
      24 | const printDocASTReducer: any = {

      at Object.visit (packages/babel-plugin-relay/node_modules/graphql/language/visitor.js:197:15)
      at print (packages/babel-plugin-relay/printGraphQL.js:21:29)
      at packages/babel-plugin-relay/__tests__/printGraphQL-test.js:27:38
```

I think our test runner does not perform that yarn install for some reason. I'm honestly not even sure why this package has its own yarn.lock.